library/x509.c

```diff
@@ -773,17 +773,26 @@ int mbedtls_x509_dn_gets( char *buf, size_t size, const mbedtls_x509_name *dn )
             ret = mbedtls_snprintf( p, n, "\?\?=" );
         MBEDTLS_X509_SAFE_SNPRINTF;
 
+        j = 0;
         for( i = 0; i < name->val.len; i++ )
         {
-            if( i >= sizeof( s ) - 1 )
+            if( i + j >= sizeof( s ) - 1 )
```

Comment:
I second @AndrzejKurek's comment on readability: Could we replace `i + j` by an `offset` variable which points to the current write-position into the target `s` buffer, and which is increased with every write? I think this make the code much easier to read.